llm_ollama.py

```diff
@@ -191,6 +205,28 @@ def build_messages(self, prompt, conversation):
         return messages
 
 
+class OllamaEmbed(llm.EmbeddingModel):
+    supports_text = True
+    supports_binary = False
+    batch_size = 8
+
+    def __init__(self, model_id):
+        self.model_id = model_id
+
+    # NOTE: this is not used, but adding it anyways
+    def __str__(self) -> str:
+        return f"Ollama: {self.model_id}"
+
+    def embed_batch(self, items):
+        result = ollama.embed(
+            model=self.model_id,
+            input=items,
+            truncate=False,  # FIXME: will error if content is too long
```

Comment:
I put not to truncate because for my use case I wanted it to always use the full text when possible. 
The fixme is because I was thinking for later that if it errored because of the chunks then try it again with a smaller chunk until you are down to one, then maybe truncate

not completely sure on this though, would be nice if there was a better way to expose the option to the user

Reply:
I changed it to truncate by default, like the default value as you said. but added a `OLLAMA_EMBED_TRUNCATE` env var option to turn it off

Reply:
Thanks, given that LLM does not implement `--option` forwarding in `llm embed` command, environment variable is the next best thing we can do. Could you please test that your implementation works though? On my side, `OLLAMA_EMBED_TRUNCATE=0 llm embed -i README.md -m mxbai-embed-large` fails because `os.getenv()` returns a string, while the `ollama.embed()` expects a boolean.

Reply:
sorry about that, fixed it

Reply:
This still does not work as you'd expect because a non-empty string becomes `True` when cast to boolean. We need to first cast to integer and only then to boolean.

However, such a conversion will support only `'1'` and `'0'` values, which is not user-friendly in the sense that seemingly valid `OLLAMA_EMBED_TRUNCATE=yes` will not work. Since we already use Pydantic library, I suggest to go for:

```python
        self.truncate = TypeAdapter(bool).validate_python(
            os.getenv("OLLAMA_EMBED_TRUNCATE", "True"),
        )
```

This supports strings that when converted to lower case are one of `'0', 'off', 'f', 'false', 'n', 'no', '1', 'on', 't', 'true', 'y', 'yes'`.

Reply:
This will error if its not a value it recognizes, so I am making it default to true if it gets an error.
But it is silent, should it print a warning?

Reply:
I think printing a warning about invalid input is a good idea.

Reply:
I added a warning.warn since print prints it multiple times. as a bonus it also tells the file/linenum